consensus/polybft/consensus_runtime.go

```diff
@@ -0,0 +1,851 @@
+	// 		return err
+	// 	}
+
+	// 	bundleProofs = append(bundleProofs,
+	// 		&BundleProof{
+	// 			Proof:      p,
+	// 			StateSyncs: events,
+	// 		})
+	// }
+
+	return c.state.insertBundles(bundleProofs)
+}
+
+// getAggSignatureForCommitmentMessage creates aggregated signatures for given commitment
+// if it has a quorum of votes
+func (c *consensusRuntime) getAggSignatureForCommitmentMessage(epoch *epochMetadata,
+	commitmentHash types.Hash) (Signature, error) {
+	validators := epoch.Validators
+
+	nodeIDIndexMap := make(map[pbft.NodeID]int, validators.Len())
+	for i, validator := range validators {
+		nodeIDIndexMap[pbft.NodeID(validator.Address.String())] = i
+	}
+
+	// get all the votes from the database for this commitment
+	votes, err := c.state.getMessageVotes(epoch.Number, commitmentHash.Bytes())
+	if err != nil {
+		return Signature{}, err
+	}
+
+	var signatures bls.Signatures
+
+	bitmap := bitmap.Bitmap{}
+
+	for _, vote := range votes {
+		index, exists := nodeIDIndexMap[vote.From]
+		if !exists {
+			continue // don't count this vote, because it does not belong to validator
+		}
+
+		signature, err := bls.UnmarshalSignature(vote.Signature)
+		if err != nil {
+			return Signature{}, err
+		}
+
+		bitmap.Set(uint64(index))
+
+		signatures = append(signatures, signature)
+	}
+
+	if len(signatures) < getQuorumSize(validators.Len()) {
+		return Signature{}, errQuorumNotReached
+	}
+
+	aggregatedSignature, err := signatures.Aggregate().Marshal()
+	if err != nil {
+		return Signature{}, err
+	}
+
+	result := Signature{
+		AggregatedSignature: aggregatedSignature,
+		Bitmap:              bitmap,
+	}
+
+	return result, nil
+}
+
+// getStateSyncEventsForBundle gets state sync events from database for the appropriate bundle
+func (c *consensusRuntime) getStateSyncEventsForBundle(from, bundleSize uint64) ([]*StateSyncEvent, error) {
+	until := bundleSize + from
+
+	return c.state.getStateSyncEventsForCommitment(from, until)
+}
+
+// startEventTracker starts the event tracker that listens to state sync events
+func (c *consensusRuntime) startEventTracker() error {
+	if c.eventTracker != nil {
+		return nil
+	}
+
+	c.eventTracker = &eventTracker{
+		config:     c.config.PolyBFTConfig,
+		subscriber: c,
+		dataDir:    c.config.DataDir,
+		logger:     c.logger.Named("event_tracker"),
+	}
+
+	if err := c.eventTracker.start(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deliverMessage receives the message vote from transport and inserts it in state db for given epoch.
+// It returns indicator whether message is processed successfully and error object if any.
+func (c *consensusRuntime) deliverMessage(msg *TransportMessage) (bool, error) {
+	epoch := c.getEpoch()
+	if epoch == nil || msg.EpochNumber < epoch.Number {
+		// Epoch metadata is undefined
+		// or received message for some of the older epochs.
+		return false, nil
+	}
+
+	if !c.isActiveValidator() {
+		return false, fmt.Errorf("validator is not among the active validator set")
```

Comment:
You can use `errors.New` here, since there is no formatting - and the error can be standardized